publisher/config/config.go

```diff
@@ -0,0 +1,135 @@
+package config
+
+import (
+	"fmt"
+	"github.com/spf13/viper"
+	"strings"
+)
+
+const (
+	defaultAptlyFolder = "/root/.aptly"
+	defaultLockgroup   = "lockgroup"
+
+	//Access points
+	accessPointStaging               = "http://nr-downloads-ohai-staging.s3-website-us-east-1.amazonaws.com"
+	accessPointTesting               = "http://nr-downloads-ohai-testing.s3-website-us-east-1.amazonaws.com"
+	accessPointProduction            = "https://download.newrelic.com"
+	mirrorProduction                 = "https://nr-downloads-main.s3.amazonaws.com"
+	placeholderAccessPointStaging    = "staging"
+	placeholderAccessPointTesting    = "testing"
+	placeholderAccessPointProduction = "production"
+)
+
+type Config struct {
+	DestPrefix           string
+	RepoName             string
+	AppName              string
+	Tag                  string
+	MirrorHost           string
+	AccessPointHost      string
+	RunID                string
+	Version              string
+	ArtifactsDestFolder  string // s3 mounted folder
+	ArtifactsSrcFolder   string
+	AptlyFolder          string
+	UploadSchemaFilePath string
+	GpgPassphrase        string
+	GpgKeyRing           string
+	AwsRegion            string
+	AwsRoleARN           string
+	// locking properties (candidate for factoring)
+	AwsLockBucket     string
+	AwsTags           string
+	LockGroup         string
+	DisableLock       bool
+	LockRetries       uint
+	UseDefLockRetries bool
+}
+
+func (c *Config) Owner() string {
+	return fmt.Sprintf("%s_%s_%s", c.AppName, c.Tag, c.RunID)
+}
+
+// parseAccessPointHost accessPointHost will be parsed to detect production, staging or testing placeholders
+// and substitute them with their specific real values. Empty will fallback to production and any other value
+// will be considered a different access point and will be return as it is
+func parseAccessPointHost(accessPointHost string) (string, string) {
+	switch accessPointHost {
+	case "":
+		return accessPointProduction, mirrorProduction
+	case placeholderAccessPointProduction:
+		return accessPointProduction, mirrorProduction
+	case placeholderAccessPointStaging:
+		return accessPointStaging, accessPointStaging
+	case placeholderAccessPointTesting:
+		return accessPointTesting, accessPointTesting
+	default:
+		return accessPointHost, accessPointHost
+	}
+}
+
+func LoadConfig() Config {
+	// TODO: make all the config required
```

Comment:
good opportunity to solve this TODO?

Reply:
I think not all of them are required in the end.